### DISCUSSION TITLE
Bind properties between parent and child component?

### DISCUSSION THREAD

**Discussion by Henriquex25** [livewire/livewire#5102] [1. Help] [answered]
I have a child component that does the customer search and I need to return the customer data found to the parent component.

If I pass the data by event, in the parent component I will not have the client instance...

So I would like to know if there is any way to link the properties between the parent component and child component?

**Comment by joshhanley** [comment, marked as answer]
@Henriquex25 currently not, the only way to share data between components is by events. But you can type hint your event listener to get it to auto-resolve the client instance for you.

```php
public function myListener(Client $client) { }
```

So when you emit your event, you just send the client ID

```php
$this->emit('myEvent', $client->id);
```
But generally I would not recommend creating a child Livewire component like that, instead I would just use a blade component for the customer search input and include it in your main component so then you don't need to share data between Livewire components.

Hope this helps!
